### DISCUSSION TITLE
Fail back bonded interface to wired connection

### DISCUSSION THREAD

**Discussion by stncttr908** [gtxaspec/wz_mini_hacks#669] [Q&A]
Hello everyone. I've got four V3s running the UCTRONICS PoE to USB adapter and for the most part they're solid, but on occasion the wired network bites the dust. The camera does not lose power; it only loses connectivity. To get around this problem I'm using the bonded network interface config and I've got some automations defined elsewhere to let me know when that failover to the WiFi interface has occurred. Is there a way to switch it back to the wired connection without rebooting the camera? It's the preferred device in the bonded pair, so I assume if there were a way to restart that interface it would assume the traffic once it was back online. Ideally I'd like to get each camera back to its wired connection as soon as possible after the failover happens since it is substantially quicker. Thanks!

**Comment by Bob-AOL** [comment]
Have you tried manually removing wi-fi availability to the cameras to see if it 'fails' back to wired? My understandinding is that the active interface will initiate dhcp discover to get back on the network regardless of which direction the 'failure' may go. I personally haven't played around with the bonding and still scratch my head over it's use case on a solid POE network where a connectivity failure would typically also include power..., until now. Silly me. LOL

FWIW, I run 8 v3s across my camera network, both wi-fi and wired (also using the UCTRONICS adapters) and just checked the 5 I have on my Blue Iris server. Seven no signal events over the past eleven days among all of them combined. Six were wi-fi with two of those being reboots.

I'm pretty sure you can also just log into the camera via SSH to switch active interfaces without a reboot. Someone with more Linux chops than I will need to jump in to explain that, tho.

**Reply by stncttr908** [reply]
Hey Bob, thanks for your reply. Yeah, it's very strange the network will drop, but not the power. I'm leaning toward some kind of driver crash, but that's just a guess since I haven't caught anything in the system or kernel logs yet to support my theory. My understanding of the bonding configuration is it will automatically switch back to the "primary" or "preferred" interface (eth0 by default) when it becomes available again:

> If you pull the RJ45-plug, ifenslave switches over to the wireless interface (e.g. wlan0) transparently, without any loss of network packages. After reconnecting the network cable, ifenslave switches back to eth0 ("failover mode").

Obviously I can't pull the RJ45 since it's also supplying power to the UCTRONICS adapter. So my only conclusion is the wired link is in fact not available, hence the question about how to reload/refresh it without a power cycle. I should also note I'm having this problem only with two of the five cameras, and it happens even after swapping out PoE/USB adapters, swapping ports on the PoE switch, etc. Luckily it seems to occur to my easily-accessible basement cameras and not the ones hanging off my siding requiring a ladder to access!

**Reply by Bob-AOL** [reply]
Double check your RJ45 terminations/connections. Power but no data points toward a physically intermittent data interruption vs. a netwrok issue. A bad kink, fastener pinch or staple will do the same thing and can drive you nuts chasing your tail on the network side.

**Reply by Bob-AOL** [reply]
Try hammering the snot of the problem cameras with some pings for awhile and see if they're dropping packets.

Try 'ping 'ip address' -t -l 1024' (for a larger than normal packet size) and just let it run. Shouldn't see much more than 10ms rtt times and zero dropped packets.

**Reply by stncttr908** [reply]
Thanks for the suggestions, I will give them a try. If I can scrounge up a long enough cable to substitute I will also try that. My thinking was if it were a cable issue it would go in and out, and cause "flapping" back and forth between the two interfaces. I don't have any evidence of that, but I am admittedly new to PoE.

**Reply by stncttr908** [reply]
So guess what, I think I am wrong in several of my assumptions. It _does_ appear to be using the wired interface based on my ping response times (the WiFi cameras are anywhere from 8-15ms and these return < 1ms) and I think having the WiFi interface listed in my mesh WiFi config is due to the shared MAC between the bonded interfaces. It shows as connected to WiFi, but cannot resolve which AP it's connected to or its IP, so it must just be detecting a MAC it knows about at Layer 2 and then getting confused. So all that said I'll probably have to come up with a new way of detecting the WiFi failover. Thanks for the help all the same.

**Reply by tdashmike** [reply]
I use BONDING_FAIL_OVER_MAC=1 so that eth0 and wlan0 have different MAC addresses.

[https://www.ibm.com/docs/en/linux-on-systems?topic=mode-option-fail-over-mac](https://www.ibm.com/docs/en/linux-on-systems?topic=mode-option-fail-over-mac)

**Reply by Bob-AOL** [reply]
With a solid network you shouldn't need wi-fi failover from a POE condition and would also need an alternate power source for 100% redundancy. That's where I can't help but scratching my head over the bonding thing. The only practical use case I can think of is if the camera is being moved from one environment to another and bonding simply boots to whichever one it detects. A member here was attempting to use it in that manner in his travels.

Also, I would designate specific APs  for wi-fi cameras instead of a mesh approach for several reasons. I would also bond the MAC addresses of each interface to specific IP addresses so your dhcp server knows what's what and why. I would imagine there's a way to clarify things for the bonding feature by minimizing its guesswork as to which hardware is really the MAC address of whatever random IP address is being assigned over whatever type of mesh configuration but again..., I don't see practical use for the bonding in my particular environment.

Keep it simple and practical with POE and failover consideration shouldn't be necessary. If your entire POE network goes down, your wi-fi network is also probably in trouble too.

But good luck with your situation whichever way you go.

If it weren' fun!

**Reply by Bob-AOL** [reply]
@tdashmike  -- bad link.

**Reply by tdashmike** [reply]
Thanks, I was using the link function incorrectly.

**Comment by robughblah** [comment]
the bit where you state “ Luckily it seems to occur to my easily-accessible basement cameras and not the ones hanging off my siding”. Possibly your basement cameras have a frequently occuring randomized power glitch, like the side effect of a freezer/furnace motor/pump etc cycling that affects the circuit they are on. try putting them on a UPS for a while. power spikes are evil. Also unplugging an ethernet cable is a layer 1 failure and hard for a device to not react to. You could create a logical failure by inserting a POE switch between and then unplugging its upstream link. RobOn Nov 18, 2023, at 9:53 PM, tdashmike ***@***.***> wrote:﻿
Thanks, I was using the link function incorrectly.

**Reply by stncttr908** [reply]
Thanks, the PoE switch is actually on a UPS already. Interesting idea for forcing a logical failure without a loss of power, appreciate the suggestion.
